frontend/packages/dev-console/integration-tests/README.md

```diff
@@ -41,48 +88,163 @@ frontend/packages/dev-console/integration-tests/
 |   |    └──helm-navigation.feature
 |   ├──  BestPractices.md   <--- Gherkin script standards
 ├── fixtures                <--- Test data required for scripts
+├── testData                <--- Test data required for scripts
 ├── cypress.json            <--- cypress configuration file
-
 ```
 
 ## This file consists of guide lines to create automation scripts
+## Designing Gherkin scripts
+
+### Feature files
+
+1. Below practices helpful to design the gherkin scripts
+   [Best Practices for writing gherkin scripts](frontend/packages/dev-console/integration-tests/features/BestPractices.md)
+2. Use the same repo for writing gherkin scripts, So that installed plugins gives you the intellisense to select the existing steps [which helps to improve re-usability]
+3. To maintain gherkin standards, linter is used [configuration file](frontend/.gherkin-lintrc) is available
+4. Execute the "yarn run gherkin-lint" command for every pr related to feature file, this helps in maintaining consistency
+5. Always comment the steps if not useful [Don't delete them]
 
 ## Scenario files
+### Generate Step Definition files
 
 1. Only validations should be present
 2. Don't use test data directly - It needs to be passed via data files (will use .ts files for now )
+1. Navigate to the "login.feature" file
+2. Select the command "Generate Steps from feature file" [By pressing "Ctrl+Shift+P" keys, commands will appear in vs code],
+3. "login" folder is generated with "login.ts" step definition file
+4. Add the comments wherever required
+5. Avoid hard coded sleep statements like cy.wait()
 
 ## View files
+### pages
 
 1. Page objects should be id, css selectors, buttontext etc.. [No XPath]- Already following
+1. Use arrow functions which helps to reduce the lines of code and it has other benefits as well
+2. Logics should be implemented within this files
+3. Don't use hard coded values [like waitTime]
+
+### page objects
+
+1. Page objects should be id, css selectors etc.. [No XPath]- Already following
 
    - Each section should have one object as shown below (It helps to reduce the import list in scenarios)
 
-     export const deleteDeployPopupObj = {
-     form: element(by.css('form.modal-content')),
-     checkbox: element(by.css('input[type="checkbox"]')),
-     cancel: element(by.css('[data-test-id="modal-cancel-action"]')),
+@@ -65,24 +114,103 @@ frontend/packages/dev-console/integration-tests/
```

Comment:
Is this diff is expected here??

Reply:
It's just an example, how to write page objects

Reply:
This `@@ -65,24 +114,103 @@...` ?? Okay, then it's up to you.

Reply:
Got it, updated it